src/engine/source/router/router.hpp

```diff
@@ -194,57 +150,60 @@ template<class Builder>
-                  "Error, type returned by Builder does not implement "
-                  "getLifter function");
-    // Assert getLifter returns a lifter
-    static_assert(
-        std::is_same_v<decltype(std::declval<builder_ret_type>().getLifter()),
-                       base::Lifter>,
-        "Error, getLifter method does not return function with signature: "
-        "std::function<Observable(Observable)>");
-
-    // Assert has a getTraceSinks method
-    static_assert(std::is_member_function_pointer_v<
-                      decltype(&builder_ret_type::getTraceSinks)>,
-                  "Error, type returned by Builder does not implement "
-                  "getTraceSinks function");
-    // Assert getTraceSinks methods returns a map<string, observable<string>>
-    static_assert(
-        std::is_same_v<
-            decltype(std::declval<builder_ret_type>().getTraceSinks()),
-            std::map<std::string, rxcpp::observable<std::string>>>,
-        "Error, getTraceSinks method does not satisfy signature: "
-        "std::function<std::map<std::string, "
-        "rxcpp::observable<std::string>>()>");
+    // // Assert Builder satisfies expected interface/functionality
+    // // First check if Builder is callable with a string as an argument
+    // static_assert(std::is_invocable_v<Builder, std::string>,
+    //               "Error, Builder type is not callable with signature: "
+    //               "Builder(std::string)");
+    // // Obtain return type of Builder call
+    // using builder_ret_type = decltype(std::declval<Builder>()(std::string {}));
+    // // Assert builder_ret_type implements the functionality needed by the router
+
+    // // Assert has a getLifter method
+    // static_assert(
+    //     std::is_member_function_pointer_v<decltype(&builder_ret_type::getLifter)>,
+    //     "Error, type returned by Builder does not implement "
+    //     "getLifter function");
+    // // Assert getLifter returns a lifter
+    // static_assert(std::is_same_v<decltype(std::declval<builder_ret_type>().getLifter()),
+    //                              base::Lifter>,
+    //               "Error, getLifter method does not return function with signature: "
+    //               "std::function<Observable(Observable)>");
+
+    // // Assert has a getTraceSinks method
+    // static_assert(
+    //     std::is_member_function_pointer_v<decltype(&builder_ret_type::getTraceSinks)>,
+    //     "Error, type returned by Builder does not implement "
+    //     "getTraceSinks function");
+    // // Assert getTraceSinks methods returns a map<string, observable<string>>
+    // static_assert(
+    //     std::is_same_v<decltype(std::declval<builder_ret_type>().getTraceSinks()),
+    //                    std::map<std::string, rxcpp::observable<std::string>>>,
+    //     "Error, getTraceSinks method does not satisfy signature: "
+    //     "std::function<std::map<std::string, "
+    //     "rxcpp::observable<std::string>>()>");
```

Comment:
Why do we keep this?

Reply:
Router is not used is this PR, is left as technical debt until API/threading definition.